gcb/presubmit.sh

```diff
@@ -68,3 +68,21 @@ bazel_run build \
 
 # Run all the tests.
 bazel_run test //src/...
+
+# Run all the tests for authorization logic with `cargo test`
+auth_logic_cargo_test() {
+    AUTH_LOGIC_DIR="rust/tools/authorization-logic"
+    INVOCATION_ID=`uuidgen`
+    STATUS_MESSAGE="Required auth-logic tests"
+    post_commit_status "${STATUS_MESSAGE}" "pending" ${INVOCATION_ID}
+    cd ${AUTH_LOGIC_DIR}
+    if cargo test
+    then
+        post_commit_status "${STATUS_MESSAGE}" "success" ${INVOCATION_ID}
+    else
+        post_commit_status "${STATUS_MESSAGE}" "failure" ${INVOCATION_ID}
+    fi
+}
+
+# Run cargo test for authorization logic
+auth_logic_cargo_test
```

Comment:
I don't see the utility in placing the auth logic portion of this script in a function and invoking it with no arguments. In other places in this script, stages are placed in functions but take some parameters so that they can be reused in different ways; this can only be used in one way. I think there are two ways you can go about fixing this:

1. Decide that this function can be generalized in a useful way and do so, passing it some arguments. For instance, this can become `cargo_test`, take a parameter that gets assigned to a variable like `CARGO_TEST_DIR` and pass `AUTH_LOGIC_DIR` as a n argument to that parameter.

2. Decide that this function is not worth generalizing and inline it into the script.

Reply:
The whole logic around posting commit status was used to extract that results URL of a bazel run. Given that we don't use bazel here, this should simply be `cargo test`. Otherwise, the post_commit_status will post an incorrect URL. 

In any case, it might be better to run this as another stage in [cloudbuild.yaml](https://github.com/google-research/raksha/blob/main/cloudbuild.yaml). Add these after the `presubmit` stage in cloudbuild.yaml and remove the code you added in presubmit.sh:
```
# Run authorization logic tests 
- id: 'auth_logic_tests'
  waitFor: ['build-image']
  name: '${_DOCKER_IMAGE}'
  entrypoint: 'bash'
  args: ['cargo', 'test']
```

These tests will run in parallel. Further, the output of the tests will aloso be nicely separated out in the google cloud build console.

Reply:
Done! :)

Reply:
Errr the fix for what mark said is done, but not what gogul said.